File: 2599-take-k-of-each-character-from-left-and-right/take-k-of-each-character-from-left-and-right.py
class Solution:
    def takeCharacters(self, s: str, k: int) -> int:
        count = [0, 0, 0]
        for i in s:
            count[ord(i) - ord('a')] += 1
        if min(count) < k:
            return -1
        result = float("inf")
        l = 0
        for r in range(len(s)):
            count[ord(s[r]) - ord('a')] -= 1

            if min(count) < k:
                count[ord(s[l]) - ord('a')] += 1
                l += 1
            
            result = min(result, len(s) - (r - l + 1))
            
        return result

File: 2599-take-k-of-each-character-from-left-and-right/take-k-of-each-character-from-left-and-right.java
class Solution {
    public int takeCharacters(String s, int k) {
        int count[] = {0, 0, 0};

        for (char c: s.toCharArray())
            count[c - 'a'] += 1;

        if (Arrays.stream(count).min().getAsInt() < k) return -1;

        int l = 0;
        int result = Integer.MAX_VALUE;
        int n = s.length();
        for (int r = 0; r < n; r++){
            count[s.charAt(r) - 'a'] -= 1;

            if (Arrays.stream(count).min().getAsInt() < k){
                count[s.charAt(l) - 'a'] += 1;
                l += 1;
            }

            result = Math.min(result, n - (r - l + 1));
        }
        return result;
    }
}